compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2772,6 +2772,46 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         Some(r)
     }
 
+    /// make sure that we are in the condition to suggest the blanket implementation, if we are in the
+    /// case of suggest it, use the function `err_reporter` to report the error or suggestion.
+    // FIXME: The copy of TyCtxt it is a bad think in this case?
+    fn maybe_lint_blanket_trait_impl<T: rustc_errors::EmissionGuarantee>(
```

Comment:
I think we can do this instead:

```suggestion
    fn maybe_lint_blanket_trait_impl<T>(
```

Reply:
Mh I think this type restriction is required to avoid the following error:

```
    Finished release [optimized] target(s) in 0.18s
Copying stage0 std from stage0 (x86_64-apple-darwin -> x86_64-apple-darwin / x86_64-apple-darwin)
Building stage0 compiler artifacts (x86_64-apple-darwin -> x86_64-apple-darwin)
   Compiling rustc_typeck v0.0.0 (/Users/vpalazzo/Documents/Github/rust/compiler/rustc_typeck)
error[E0277]: the trait bound `T: EmissionGuarantee` is not satisfied
    --> compiler/rustc_typeck/src/astconv/mod.rs:2781:26
     |
2781 |         diagnostic: &mut DiagnosticBuilder<'_, T>,
     |                          ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EmissionGuarantee` is not implemented for `T`
     |
note: required by a bound in `DiagnosticBuilder`
    --> /Users/vpalazzo/Documents/Github/rust/compiler/rustc_errors/src/diagnostic_builder.rs:21:37
     |
21   | pub struct DiagnosticBuilder<'a, G: EmissionGuarantee> {
     |                                     ^^^^^^^^^^^^^^^^^ required by this bound in `DiagnosticBuilder`
help: consider restricting type parameter `T`
     |
2778 |     fn maybe_lint_blanket_trait_impl<T: rustc_errors::EmissionGuarantee/*: rustc_errors::EmissionGuarantee*/>(
     |                                       +++++++++++++++++++++++++++++++++

For more information about this error, try `rustc --explain E0277`.
error: could not compile `rustc_typeck` due to previous error
Build completed unsuccessfully in 0:00:05
```